From 58a5fcf24b76c3cf80d671c215b59a0fef76a5c5 Mon Sep 17 00:00:00 2001
Message-Id: <58a5fcf24b76c3cf80d671c215b59a0fef76a5c5.1387384437.git.minovotn@redhat.com>
From: Miroslav Rezanina <mrezanin@redhat.com>
Date: Wed, 20 Nov 2013 07:51:51 +0100
Subject: [PATCH 1/2] Install systemd services into /lib/systemd instead of
 /usr/lib64/systemd

RH-Author: Miroslav Rezanina <mrezanin@redhat.com>
Message-id: <6bbcb58399842495e8e42b2473c7e1c5974bca1d.1384931889.git.mrezanin@redhat.com>
Patchwork-id: 55781
O-Subject: [RHEL7 qemu-kvm PATCHv2 3/3] Install systemd services into /lib/systemd instead of /usr/lib64/systemd
Bugzilla: 1025217
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Michal Novotny <minovotn@redhat.com>

From: Miroslav Rezanina <mrezanin@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1025217
Systemd is not able to find services in /usr/lib64/systemd directory.
Moving ksm related services to proper directory.

v2:
Used different location than Fedora.
Unit files (ksm.service and ksmtuned.service) are stored in
/usr/lib/systemd/system as this is directory where unit files installed
by rpm should go (handled by _unitdir variable).
ksmctl binary is stored to /usr/libexec as this is directory more suitable for
it. ksm.service is modified to reflect this change.

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 redhat/ksm.service            |  4 ++--
 redhat/qemu-kvm.spec.template | 12 ++++++------
 2 files changed, 8 insertions(+), 8 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 redhat/ksm.service            |  4 ++--
 redhat/qemu-kvm.spec.template | 12 ++++++------
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/redhat/ksm.service b/redhat/ksm.service
index ccc2767..35c6f1d 100644
--- a/redhat/ksm.service
+++ b/redhat/ksm.service
@@ -6,8 +6,8 @@ ConditionPathExists=/sys/kernel/mm/ksm
 Type=oneshot
 RemainAfterExit=yes
 EnvironmentFile=-/etc/sysconfig/ksm
-ExecStart=/lib/systemd/ksmctl start
-ExecStop=/lib/systemd/ksmctl stop
+ExecStart=/usr/libexec/ksmctl start
+ExecStop=/usr/libexec/ksmctl stop
 
 [Install]
 WantedBy=multi-user.target
diff --git a/redhat/qemu-kvm.spec.template b/redhat/qemu-kvm.spec.template
index 9220681..6c9ecc0 100644
--- a/redhat/qemu-kvm.spec.template
+++ b/redhat/qemu-kvm.spec.template
@@ -490,11 +490,11 @@ dobuild --target-list="$buildarch"
 %define _udevdir %(pkg-config --variable=udevdir udev)/rules.d
 
 %if 0%{!?build_only_sub:1}
-    install -D -p -m 0644 %{SOURCE4} $RPM_BUILD_ROOT%{_libdir}/systemd/system/ksm.service
+    install -D -p -m 0644 %{SOURCE4} $RPM_BUILD_ROOT%{_unitdir}/ksm.service
     install -D -p -m 0644 %{SOURCE5} $RPM_BUILD_ROOT%{_sysconfdir}/sysconfig/ksm
-    install -D -p -m 0755 ksmctl $RPM_BUILD_ROOT%{_libdir}/systemd/ksmctl
+    install -D -p -m 0755 ksmctl $RPM_BUILD_ROOT%{_libexecdir}/ksmctl
 
-    install -D -p -m 0644 %{SOURCE7} $RPM_BUILD_ROOT%{_libdir}/systemd/system/ksmtuned.service
+    install -D -p -m 0644 %{SOURCE7} $RPM_BUILD_ROOT%{_unitdir}/ksmtuned.service
     install -D -p -m 0755 %{SOURCE8} $RPM_BUILD_ROOT%{_sbindir}/ksmtuned
     install -D -p -m 0644 %{SOURCE9} $RPM_BUILD_ROOT%{_sysconfdir}/ksmtuned.conf
 
@@ -712,10 +712,10 @@ sh %{_sysconfdir}/sysconfig/modules/kvm.modules &> /dev/null || :
     %{_mandir}/man1/%{pkgname}.1*
     %attr(4755, -, -) %{_libexecdir}/qemu-bridge-helper
     %config(noreplace) %{_sysconfdir}/sasl2/%{pkgname}.conf
-    %{_libdir}/systemd/system/ksm.service
-    %{_libdir}/systemd/ksmctl
+    %{_unitdir}/ksm.service
+    %{_libexecdir}/ksmctl
     %config(noreplace) %{_sysconfdir}/sysconfig/ksm
-    %{_libdir}/systemd/system/ksmtuned.service
+    %{_unitdir}/ksmtuned.service
     %{_sbindir}/ksmtuned
     %config(noreplace) %{_sysconfdir}/ksmtuned.conf
     %dir %{_sysconfdir}/%{pkgname}
-- 
1.7.11.7

